apps/mark-scan/frontend/src/pages/ballot_invalidated_page.tsx

```diff
@@ -0,0 +1,53 @@
+import { useContext } from 'react';
+
+import { InsertedSmartCardAuth } from '@votingworks/types';
+
+import { confirmInvalidateBallot } from '../api';
+import { BallotContext } from '../contexts/ballot_context';
+import { AskPollWorkerPage } from './ask_poll_worker_page';
+
+interface Props {
+  authStatus:
+    | InsertedSmartCardAuth.CardlessVoterLoggedIn
+    | InsertedSmartCardAuth.PollWorkerLoggedIn;
+}
+
+export function BallotInvalidatedPage({ authStatus }: Props): JSX.Element {
+  const { resetBallot, endVoterSession } = useContext(BallotContext);
+
+  const confirmInvalidateBallotMutation = confirmInvalidateBallot.useMutation();
+
+  async function onPressContinue() {
+    // Reset session and ballot before changing the state machine state. If done
+    // in the reverse order the screen will flicker
+    await endVoterSession();
+    resetBallot();
+    confirmInvalidateBallotMutation.mutate(undefined);
+  }
+
+  return (
+    <AskPollWorkerPage
```

Comment:
Is there a reason that this flow is abstracted out into a separate component? It seems like it's only used here in `BallotInvalidatedScreen`

Reply:
Great question, I use it in this branch that follows the same pattern but for blank sheet interpretations:
https://github.com/votingworks/vxsuite/pull/3981/files#diff-1db666e555eff795a25c5b430171548b3cbea77dec73e069948df2552e78376e